Deploy terminals by USB (#170)
uyuni-project/uyuni-docs: modules/retail/pages/retail-deploy-terminals.adoc

----
HWTYPE:POSVendor-Terminal1
----
. In the {susemgr} {webui}, navigate to menu:Systems[System Groups], and click the btn:[Create Group] button.

> . In the [guimenu]``Create System Group`` dialog, create a new system group, using the hardware type group name you created earlier.

Reviewer: ```suggestion
. In the [guimenu]``Create System Group`` dialog, create a new system group, using the hardware type group name you determined in step one of this procedure.
```

> 

Reviewer: ```suggestion
```

== Deploy Images to the Terminals

When you have your bootstrap image ready, you can deploy the image to the terminals.

.Procedure: Deploying Images to the Terminals

. Power on your POS terminals.
. The branch server will bootstrap the terminals according to the data you have provided.

== Deploy Terminals with a Removable USB Device

If you do not want to boot terminals from the network, you can create a live USB image and deploy terminals using a removable USB storage device.

> This is useful if you are unable or unwilling to connect some or all of your terminals to the network, or if you do not have a local {smr} branch server providing network services.

Reviewer: ```suggestion
This is useful if you cannot connect your terminals to the network, or if you do not have a local {smr} branch server providing network services.
```
The user will know why USB might be a good option.

> This is useful if you are unable or unwilling to connect some or all of your terminals to the network, or if you do not have a local {smr} branch server providing network services.

Reviewer 2: This is not entirely connect. Terminals still need access to the network because the USB stick contains only boot image part usually served by PXE. The system image with OS and custom retail applications still needs to be fetched from SUMA server.

I think that use case for this feature are ATMs. They have network connection, but they do not have local branch which would provide PXE and tftp services. So they are booted using this USB stick (which may contain VPN configuration) and connect to some remote branch in the datacenter already through salt.

You can prepare a bootable USB device with the image and tools required to deploy a POS terminal using a remote {smr} branch server.
POS devices booted using the USB device are assigned to the {smr} branch server that created the USB device.

You can create the bootable USB device on the branch server directly, or on the {smr} Server.

.Procedure: Creating a Bootable USB Device on the Branch Server
. On the {smr} branch server, at the command prompt, as root, create the POS image:
+
----

> salt-call image_sync_usb.create <usb image name> <size in MiB>

Reviewer 2: I forgot to add that the size should be at least 300MB.

----
. Insert the USB device into the {smr} branch server machine, and copy the image to the new location:
+
----
dd if=<usb image name> of=<path to usb device>
----

.Procedure: Creating a Bootable USB Device on the {smr} Server
. On the {smr} Server, at the command prompt, as root, create the POS image:
+
----
salt <branch_server> image_sync_usb.create <usb image name> <size in MiB>
----
. Insert the USB device into the {smr} Server machine, and copy the image to the new location:
+
----
salt <branch_server> cmd.run "dd if=<usb image name> of=<path to usb device on branch>"

> ----

Reviewer 2: This is not correct. The snipped was meant to show that user can trigger usb stick build from server, however the resulting image is still on branch server.
So unless user transfers the image from branch server to suma server, usb device needs to be available on branch server.
I think this section brings only more confusion and probably best is to remove this procedure and leave only the first one where the image is created on branch server.

> 

Reviewer: ```suggestion
```

When you have the image on the USB drive, check that the terminals you want to deploy allow local booting.
You can check this by editing the Saltboot formula in the {smr} {webui}.
For more information about the Saltboot formula, see xref:salt:formula-saltboot.adoc[].

.Procedure: Deploying Images to the Terminals using USB

. Power on the POS terminal.
. Insert the USB device into the terminal.

> . If not automatically done, mount the USB drive to begin bootstrapping.

Reviewer 2: This USB is bootable, so the order should be:
```adoc
. Insert the USB device into the terminal.
. Power on the POS terminal.
. Make sure booting from USB device is enabled and boot from USB device.
```
